### PR TITLE
Update package

### PR DESCRIPTION
**Pull request by pluflou**
move to pyproject.toml
clean up dependencies
remove mosek from test dependencies
update ci/cd, builds and conda recipe

### CODE REVIEW

**Comment by pluflou** [comment]
This is not ready for merging yet. The remaining to-dos are:

- [ ] fix test CI to run when PR is triggered from forks
- [ ] add optional deps in TOML for docs, tests and mosek
- [ ] consider removing requirements.txt to minimize redundancy
- [ ] add deprecation warning to statistical clear sky package
- [ ] resolve future warnings

File: .github/workflows/test.yml
Comment (pluflou) [inline .github/workflows/test.yml:1]: revert to original

**Comment by pluflou** [comment]
@bmeyers this is ready for review/approval. Note that I edited the test.yml to bring in the work  @nimish did now so that the test workflow runs at PR to main (it was only running on push to the working branch in my fork). Nimish is adding a fix to get that to run for both without duplicates when there is a PR open from a branch on the main repo. I also added his updates to expand the tests and pretty up the output.

**Review by bmeyers — APPROVED** [review]

**Comment by bmeyers** [comment]
@pluflou just approved, but It appears that merging is blocked by a failed test deployment.

**Comment by pluflou** [comment]
> @pluflou just approved, but It appears that merging is blocked by a failed test deployment.

Oh oops. I think it's because I changes the test workflow names and it's looking for the old one. Let me fix that.

**Pull request merged**